pkg/agent/openflow/client.go

```diff
@@ -434,6 +440,12 @@ func (c *client) GetPodFlowKeys(interfaceName string) []string {
 func (c *client) InstallServiceGroup(groupID binding.GroupIDType, withSessionAffinity bool, endpoints []proxy.Endpoint) error {
 	c.replayMutex.RLock()
 	defer c.replayMutex.RUnlock()
+
+	if len(endpoints) > maxEndpoints {
+		endpoints = endpoints[:maxEndpoints]
+		klog.Warningf("In group %v, Endpoints of current Service exceeds %d, and extra Endpoints will be dropped.", groupID, maxEndpoints)
```

Comment:
Group ID is not very helpful to understand which service it is. Could we move it to `proxier.installServices` to get a more friendly context.
Besides, I just found cutting endpoints here will cause inconsitency between the group and the flows that are installed by `InstallEndpointFlows` and the cache `proxier.endpointsInstalledMap`. Not sure if it will cause some issues, to be safe, should we cut it earlier to avoid this risk?

Reply:
Not sure how often the log will be printed in practice, better to have some mechanim to avoid same logs overwhelming log files, for example, only log a service once if it keeps exceeding 800 endpoints (for example, use a strings.Set to keep the logged services and remove a service from it when it has less than 800 endpoints).

Reply:
Agreed, it's better to cut it earlier.

Reply:
+1 to Quan's suggestions
these are good points